homeassistant/components/denonavr/__init__.py

```diff
@@ -57,6 +64,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    async def disconnect(event: Event) -> None:
+        """Disconnect from Telnet."""
+        if (
+            entry.options.get(CONF_USE_TELNET, DEFAULT_USE_TELNET)
+            and receiver is not None
+        ):
+            await receiver.async_telnet_disconnect()
+
+    if entry.options.get(CONF_USE_TELNET, DEFAULT_USE_TELNET):
+        entry.async_on_unload(
+            hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, disconnect)
```

Comment:
```suggestion
            hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_disconnect)
```

Reply:
Addressed in https://github.com/home-assistant/core/pull/85980/commits/f5e8b057d1716a236dcc1062734a5688d40a725e